### PR TITLE
Uniquify generated exercise and section names

### LINKED ISSUE #458: Shapeless Tuples exercises in website is pointing to Stdlib Tuples exercises
Shapeless Tuples exercises in the website show you, the stdlib tuples exercises. Also, Github links of this page are broken, since it redirects you to https://github.com/scala-exercises/exercises-shapeless/blob/master/src/main/scala/stdlib/Tuples.scala.

### PR DESCRIPTION
**Pull request by dialelo**
And change the target package from `defaultLib` for the more semantic
`scalaExercisesContent`. Closes #458

### CODE REVIEW

**Comment by dialelo** [comment]
@raulraja please take a look!

**Comment by raulraja** [comment]
LGTM!

**Comment by dialelo** [comment]
Now that it's green I'm going to open pull requests to the exercises so they get released, I'll update this PR as soon as they are released and bumped.

**Comment by andyscott** [comment]
Is there any way to rebase and fix all the commits so we are green all the way through?

**Comment by dialelo** [comment]
>Is there any way to rebase and fix all the commits so we are green all the way through?

:+1: i'll do that when tests pass.

**Comment by dialelo** [comment]
Squashed everything into one commit as per @andyscott's suggestion, @raulraja should be ready to merge when the tests pass.

**Comment by raulraja** [comment]
Awesomesauce!

**Comment by raulraja** [comment]
The merge to master seems to be failing: https://travis-ci.org/scala-exercises/scala-exercises/builds/140613073
```
[info] [error] Unable to find exercise compiler class

[info] java.lang.ClassNotFoundException: com.fortysevendeg.exercises.compiler.CompilerJava
```

**Comment by andyscott** [comment]
@raulraja I think this is because the version number was changed to a non snapshot version-- so now it isn't picking up the locally published artifacts for the compiler (so they never end up on the plugin's classpath).

File: build.sbt [hunk 1]
```diff
@@ -31,7 +31,7 @@ lazy val formattingSettings = SbtScalariform.scalariformSettings ++ Seq(
 
 lazy val commonSettings = Seq(
   organization := "org.scala-exercises",
-  version := "0.0.0-SNAPSHOT",
+  version := "0.1.0",
```
Comment (andyscott) [inline build.sbt]: This change is what I'm referring to.

**Pull request merged**